### PR TITLE
Schedule

### PR DESCRIPTION
**Pull request by TakenPilot**
New Feature
- Add `/schedule` endpoint that:
  - accepts POST at `/schedule` with `{at: <timestamp>, publish: <page or component uri>}` returning `{at: <timestamp>, publish: <page or component uri>, _ref: <new id uri>}`
  - accepts GET at `/schedule` and `/schedule/:id`
  - accepts DELETE at `/schedule/:id`

### CODE REVIEW

**Comment by yoshokatana** [comment]
Yaaaaay. Is there a GET endpoint(s)?

**Comment by TakenPilot** [comment]
Sure is!

File: lib/services/db.js [hunk 1]
```diff
@@ -33,35 +33,51 @@ function defer() {
...
+ */
+function pipeToPromise(pipe) {
+  var str = '',
+    d = defer();
+
+  pipe.on('data', function (data) { str += data; })
+    .on('error', d.reject)
+    .on('end', function () { d.resolve(str); });
+  return d.promise;
+}
```
Comment (cruzanmo) [inline lib/services/db.js]: nice

**Comment by yoshokatana** [comment]
:+1: but add docs?

File: lib/services/references.js [hunk 2]
```diff
@@ -24,7 +29,7 @@ function replaceVersion(uri, version) {
 
 /**
  * @param {string} version
- * @returns {Function}
+ * @returns {function}
```
Comment (cruzanmo) [inline lib/services/references.js]: Ha, I believe it is truly an instance of Function with a capital F, but consistency with the other JSDocs is good.

File: lib/services/schedule.js [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+  components = require('../services/components'),
+  db = require('../services/db'),
+  references = require('../services/references'),
+  siteService = require('../services/sites'),
+  scheduledAtProperty = 'at',
+  publishProperty = 'publish',
+  pages = require('../services/pages'),
+  log = require('../log');
+
+function setScheduleInterval(value) {
```
Comment (cruzanmo) [inline lib/services/schedule.js]: @param

**Comment by cruzanmo** [comment]
+1

File: lib/services/references.js (hunk 2 above)
Comment (TakenPilot) [inline lib/services/references.js]: `Function` is a class, `function` is a native type.  : )

**Comment by yoshokatana** [comment]
:+1:

**Pull request merged**